File: Selenium/Activities/Actiivty4.py
from selenium import webdriver
from selenium.webdriver.common.by import By

# Set up the driver
driver = webdriver.Chrome()

# Open the target practice page
driver.get("https://training-support.net/webelements/target-practice")

# Print the title of the page
print("Page title:", driver.title)

# Using XPath: Find the 3rd header and print its text
third_header = driver.find_element(By.XPATH, "//h3")
print("3rd header text:", third_header.text)

# Using XPath: Find the 5th header and print its color
fifth_header = driver.find_element(By.XPATH, "//h5")
print("5th header color:", fifth_header.value_of_css_property("color"))

# Using another locator: Find the purple button and print all its classes
purple_button = driver.find_element(By.CLASS_NAME, "purple")
print("Purple button classes:", purple_button.get_attribute("class"))

# Using another locator: Find the slate button and print its text
slate_button = driver.find_element(By.CLASS_NAME, "slate")
print("Slate button text:", slate_button.text)

# Close the browser
driver.quit()

File: Selenium/Activities/Actiivty4.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;

public class Activity4 {
    public static void main(String[] args) {
        // Set up the driver
        WebDriver driver = new ChromeDriver();

        // Open the target practice page
        driver.get("https://training-support.net/webelements/target-practice");

        // Print the title of the page
        System.out.println("Page title: " + driver.getTitle());

        // Using XPath: Find the 3rd header and print its text
        WebElement thirdHeader = driver.findElement(By.xpath("//h3"));
        System.out.println("3rd header text: " + thirdHeader.getText());

        // Using XPath: Find the 5th header and print its color
        WebElement fifthHeader = driver.findElement(By.xpath("//h5"));
        System.out.println("5th header color: " + fifthHeader.getCssValue("color"));

        // Using another locator: Find the purple button and print all its classes
        WebElement purpleButton = driver.findElement(By.className("purple"));
        System.out.println("Purple button classes: " + purpleButton.getAttribute("class"));

        // Using another locator: Find the slate button and print its text
        WebElement slateButton = driver.findElement(By.className("slate"));
        System.out